src/clojure_elastic_apm/ring.clj

```diff
@@ -2,11 +2,44 @@
   (:require [clojure-elastic-apm.core :refer [with-apm-transaction
                                               type-request]]))
 
-(defn wrap-apm-transaction [handler]
-  (fn [{:keys [request-method uri] :as request}]
-    (let [tx-name (str (.toUpperCase (name request-method)) " " uri)]
-      (with-apm-transaction [tx {:name tx-name :type type-request}]
-        (let [{:keys [status] :as response} (handler (assoc request :clojure-elastic-apm/transaction tx))]
-          (when status
-            (.setResult tx (str "HTTP " status)))
-          response)))))
+(defn match-uri [pattern uri]
+  (let [pattern-segs (clojure.string/split pattern #"/")
+        uri-segs (clojure.string/split uri #"/")
+        matcher (fn [p u]
+                  (cond
+                    (= p "*") u
+                    (= p "_") "*"
+                    (= p u) u
+                    :else false))]
+    (if (> (count pattern-segs)
+           (count uri-segs))
+      false
+      (let [matches (map matcher pattern-segs uri-segs)
+            matched? (reduce #(and %1 %2) matches)]
```

Comment:
`kibit` suggests the following:
```
Consider using:
  and
instead of:
  #(and %1 %2)
```

Reply:
Kibit is wrong here. :D `and` is a macro, you can't pass it on its own as if it were a function.